tubular/dates.py

```diff
@@ -1040,20 +1040,56 @@ def __init__(
                 )
             )
 
-        if not isinstance(units, str):
+        if not isinstance(units, str) and not isinstance(units, dict):
             raise TypeError(
-                "{}: units must be a string but got {}".format(
+                "{}: units must be a string or dict but got {}".format(
                     self.classname(), type(units)
                 )
             )
 
-        if (not isinstance(period, int)) and (not isinstance(period, float)):
+        if (
+            (not isinstance(period, int))
+            and (not isinstance(period, float))
+            and (not isinstance(period, dict))
+            or (isinstance(period, bool))
```

Comment:
explicit check for bool dtype unnescessary?

Reply:
apparently bool is a subclass of int so if a bool entered it sneaks through - did in the tests i made anyway